components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/SynapseAnalyticsDataProvider.java

```diff
@@ -323,12 +328,15 @@ public Map<String, Object> getProperties() {
         } else {
             customProperties = new HashMap<>();
         }
-        customProperties.put(Constants.API_USER_NAME_KEY, getUserName());
+        customProperties.put(Constants.API_USER_NAME_KEY, getEndUserName());
         customProperties.put(Constants.API_CONTEXT_KEY, getApiContext());
         return customProperties;
     }
 
-    private String getUserName() {
+    // previously was getUserName
+    // with the modifications in AnalyticsDataProvider
+    // there will be access modifier clash, so renamed as getEndUserName
+    private String getEndUserName() {
```

Comment:
why can't u move this content within getUserName method itself rather than returning getEndUserName()?

Reply:
Yes Tishan, there is no issue doing that..
Will modify accordingly..